src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -533,7 +534,7 @@ private User impersonate(final RestRequest request, final User originalUser) thr
             //loop over all http/rest auth domains
             for (final AuthDomain authDomain: restAuthDomains) {
                 final AuthenticationBackend authenticationBackend = authDomain.getBackend();
-                if (authDomain.getBackend() instanceof NoOpAuthenticationBackend  ) {
+                if (authDomain.getHttpAuthenticator() instanceof OnBehalfOfAuthenticator) {
```

Comment:
This doesn't seem right, the auth domain associated with the OnBehalfOfAuthenticator isn't configured correctly

Reply:
I think this is the solution I can think of for only skipping the OBO auth backend but not the other backends.(cc: @cwperks )